### PR TITLE
boards: shields: Add Atmel OLED1 Xplained Pro

### PR DESCRIPTION
**Pull request by atomsmith**
The OLED1 Xplained Pro shield provides a monochrome 128x32 OLED display as well as three discrete indicator LEDs and three momentary pushbutton switches. No new driver code is required; the SPI-connected display is already supported by the ssd1306 driver.

This PR has been tested on three Atmel Xplained Pro evaluation boards: SAM D21, SAM E70, and SAM V71. Any of the compatible extension headers on the boards can be used.

### CODE REVIEW

File: boards/shields/atmel_oled1_xpro/Kconfig.shield [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2023 The Zephyr Project Contributors
```
Comment (erwango) [inline boards/shields/atmel_oled1_xpro/Kconfig.shield:1]: @kartben, Is this copyright valid in current context of individual contribution ?

Comment (henrikbrixandersen) [reply boards/shields/atmel_oled1_xpro/Kconfig.shield:1]: This form is generally not used in the Zephyr project.

File: CODEOWNERS
Comment (henrikbrixandersen) [inline CODEOWNERS:1]: In general there is no need to update the CODEOWNERS file. Its use will be deprecated soon.

File: boards/arm/atsamd21_xpro/atsamd21_xpro-pinctrl.dtsi [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 2022, Gerson Fernando Budke <nandojve@gmail.com>
+ * Copyright (c) 2023 Scott Barnes <scott@atomsmith.net>
```
Comment (henrikbrixandersen) [inline boards/arm/atsamd21_xpro/atsamd21_xpro-pinctrl.dtsi:3]:
```suggestion
 * Copyright (c) 2024 Scott Barnes <scott@atomsmith.net>
```

File: boards/arm/atsamd21_xpro/atsamd21_xpro.dts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 2018 Bryan O'Donoghue
+ * Copyright (c) 2023 Scott Barnes <scott@atomsmith.net>
```
Comment (henrikbrixandersen) [inline boards/arm/atsamd21_xpro/atsamd21_xpro.dts:3]:
```suggestion
 * Copyright (c) 2024 Scott Barnes <scott@atomsmith.net>
```

File: boards/arm/atsamd21_xpro/atsamd21_xpro.dts [hunk 4]
```diff
@@ -132,3 +208,36 @@ zephyr_udc0: &usb0 {
...
+};
+
+ext3_spi: &sercom5 {
+};
+
+ext3_i2c: &sercom2 {
+};
+
+ext3_serial: &sercom4 {
+};
```
Comment (henrikbrixandersen) [inline boards/arm/atsamd21_xpro/atsamd21_xpro.dts:243]: If pins are not assigned to these, how can they be routed to a specific shield function?

File: boards/arm/sam_e70_xplained/sam_e70_xplained-common.dtsi [hunk 5]
```diff
@@ -3,6 +3,7 @@
  * Copyright (c) 2017 Justin Watson
  * Copyright (c) 2020 Stephanos Ioannidis <root@stephanos.io>
  * Copyright (c) 2020 Gerson Fernando Budke <nandojve@gmail.com>
+ * Copyright (c) 2023 Scott Barnes <scott@atomsmith.net>
```
Comment (henrikbrixandersen) [inline boards/arm/sam_e70_xplained/sam_e70_xplained-common.dtsi:6]:
```suggestion
 * Copyright (c) 2024 Scott Barnes <scott@atomsmith.net>
```

File: boards/shields/atmel_oled1_xpro/atmel_oled1_xpro.overlay [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+		zephyr,display = &atmel_oled1_xpro;
+	};
+
+	aliases {
+		oled-led1 = &oled1_led1;
+		oled-led2 = &oled1_led2;
+		oled-led3 = &oled1_led3;
+		oled-sw1 = &oled1_button1;
+		oled-sw2 = &oled1_button2;
+		oled-sw3 = &oled1_button3;
```
Comment (henrikbrixandersen) [inline boards/shields/atmel_oled1_xpro/atmel_oled1_xpro.overlay:18]: These aliases are non-standard and seems shield-specific. Where are they used?

File: boards/shields/atmel_oled1_xpro/atmel_oled1_xpro.overlay [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+	};
+
+	oled1_gpio_keys {
```
Comment (henrikbrixandersen) [inline boards/shields/atmel_oled1_xpro/atmel_oled1_xpro.overlay:21]: You need to use the full shield name as nodelabel prefix all over.

File: boards/shields/atmel_oled1_xpro/atmel_oled1_xpro.overlay [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+			label = "OLED Board LED 2";
+		};
+		oled1_led3: led_3 {
+			gpios = <&ext1_header 3 GPIO_ACTIVE_LOW>;
+			label = "OLED Board LED 3";
+		};
+	};
+};
+
+&ext1_spi {
```
Comment (henrikbrixandersen) [inline boards/shields/atmel_oled1_xpro/atmel_oled1_xpro.overlay:58]: Non-standardized nodelabel.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

**Pull request closed without merging**